src/main/java/com/salesforce/dynamodbv2/mt/cache/MtCache.java

```diff
@@ -25,19 +25,46 @@
  *
  * @author msgroi
  */
-public class MtCache<V> implements Cache<String, V> {
+public class MtCache<V> implements Cache<Object, V> {
+
+    private static class Key {
```

Comment:
Note: this is needed both for performance and correctness (since current delimiter-based approach can result in collisions).